include/mitsuba/python/python.h

```diff
@@ -63,6 +63,15 @@ PYBIND11_DECLARE_HOLDER_TYPE(T, mitsuba::ref<T>, true);
             (void) new Class(                                                  \
                 name, #Name, variant,                                          \
                 [=](const Properties &p) {                                     \
+                    /* The thread-local python variant information might not
+                    have been set on this thread */                            \
+                    py::gil_scoped_acquire gil;                                \
```

Comment:
Looks good. I would add a nested scope to ensure that we only hold the GIL for the `py::*` operations, but not the subsequent plugin construction.

Reply:
@wjakob I don't think this works. When I tested it, it seemed like the the local thread state was being deleted when the GIL was released.
 
 This line in Pybind seems to confirm it: https://github.com/pybind/pybind11/blob/f9f00495a3f0ef6037c215c42bd2d919590ff11f/include/pybind11/gil.h#L101
 
Should I try to workaround this with the C API directly?

Reply:
https://stackoverflow.com/questions/26061298/python-multi-thread-multi-interpreter-c-api